rfcs/text/0000-image-base-transcriptomics.md

```diff
@@ -0,0 +1,497 @@
+complete dataset, in a cloud-based environment.
+
+#### Output Data Formats
+
+Image-based transcriptomics experiments emit three output formats that
+contain identified features, all of which can be emitted in either
+netcdf or zarr file formats, the latter being the format consumed by the
+matrix service. In addition, it has the opportunity to emit image files
+that have been processed to align images, remove autofluorescence and
+enhance spots. All output formats (including SpaceTx image format)
+contain full provenance records of the analysis environment, versions of
+starfish and key dependencies, algorithms that were run, and their
+parameters.
+
+##### [IntensityTable]
+
+Image-based transcriptomics experiments image the same tissue using
+multiple color channels, and after subjecting the tissue to different
+reagents. The physical locations of fluorescent molecules manifest as
+spots in the images, and these spots are detected and aggregated across
+fluorescence channels and time to build codes. In practice, the process
+of detecting and decoding spots contains significant possibilities for
+error, and the IntensityTable supports analysis and QC for this
+process.
+
+##### [SegmentationMask]
+
+In order to assign the spots identified above to individual cells, the
+locations of cells within the images must be identified. Compressed
+segmentation masks are a compact way to represent these data, which
+store, for each cell, the portion of the image that is attributed to
+that cell.
+
+##### [ExpressionMatrix]
+
+*starfish* combines spot calls from the IntensityTable with cell areas
+in the SegmentationMask to produce expression matrices. These are saved
+in zarr format and look identical to the matrices emitted by the Data
+Analysis Service pipelines, with the exception that each cell is
+additionally annotated with metadata that specify the x, y, and z
+location of the data.
+
+##### Processed Images
+
+Image-based transcriptomics experiments suffer from several sources of
+error and variation which are removed via image processing. Microscope
+stages or tissue slices may shift as reagents are added and removed
+across imaging rounds, or which spectral filters are swapped between
+channels. In addition, reagents can non-specifically bind to cellular
+material, causing autofluorescence. Prior to spot-calling, images can be
+aligned, background removed, and spots computationally enhanced.
+*starfish* can output these processed images, annotated with information
+sufficient to re-do the processing, so that the images input to spot
+calling can be evaluated by users. If emitted, processed images will be
+generated in SpaceTx Format.
+
+#### HCA Sample Metadata
+
+The HCA metadata team has built a metadata specification draft for
+imaging data that references data in SpaceTx-Format. The issue that
+formed the draft can be found
+[<span class="underline">here</span>](https://github.com/HumanCellAtlas/metadata-schema/issues/368).
```

Comment:
Requirements here included:
- Generality for experiments including immunohistochemistry, in situ sequencing and other FISH methods.
- Multiplexed and non-multiplexed experiments.
- Target to probe mapping
- Target level metadata (search by gene)
- Congruence with current entity model
- Explicit link to experiment.json which links to all other data files.
- One experiment.json per assay process (aka bundle)

Reply:
> * One experiment.json per assay process (aka bundle)

Is "assay process" consistent with a "replicate" as we think of in sequencing? There may be multiple `experiment.json` files per study. 

Everything else looks good. 👍

Reply:
This sounds like a good feature request for starfish. Making sure I understand, the key objects the HCA needs are: 

1. `experiment.json` (top-level project)
2. `run.json` (a complete collage of a tissue sample containing many fields of views and potentially multiple imaging modalities (e.g. DIC))
3. `field_of_view.json` (a piece of the collage)

Our manifest.json could probably be repurposed to store the `run.json` If that's right I'll make an issue and reference it here.

Reply:
Zina's suggestion is fine with me, though there are multiple solutions. At the interface between spacetx metadata and hca metadata the main requirement is good definitions on entities that contributors and wranglers can abide by. 'experiment.json' is too loosely defined at the moment (maybe by design) to be relied upon to be consistent across datasets. In this informational RFC I think it is worth adding some detail to this section as I listed above and ensuring that our definition of experiment is clearly defined.

Reply:
@ambrosejcarr I suggest adding: 
1. Metadata design requirements as bulleted above (maybe these could be written in a clearer more accessible way)
2. SpaceTx entity definitions of experiment, FOV etc. as these affect how hca metadata is linked to the data. Changing these definitions would need us to revisit how we link data to metadata.